app/controllers/congress_messages_controller.rb

```diff
@@ -0,0 +1,101 @@
+class CongressMessagesController < ApplicationController
+  include Tooling
+
+  before_action :set_congress_message_campaign
+  before_action :update_user, only: :create
+  before_action :subscribe_user, only: :create
+
+  rescue_from SmartyStreets::AddressNotFound, with: :address_not_found
+  rescue_from CongressForms::RequestFailed, with: :congress_forms_request_failed
+
+  def new
+    if @campaign.target_bioguide_ids.present?
+      bioguide_ids = @campaign.target_bioguide_ids.split
+    else
+      location = SmartyStreets.get_location(params["street_address"], params["zipcode"])
+      members = @campaign.targets.for_district(location.state, location.district)
+      bioguide_ids = members.pluck(:bioguide_id)
+    end
+
+    forms = CongressForms::Form.find(bioguide_ids)
+    @message = CongressMessage.new_from_lookup(location, params[:message], @campaign, forms)
+    render partial: "form"
+  end
+
+  def create
+    @message = CongressMessage.new(congress_message_params.merge(campaign: @campaign))
+    @message.forms = CongressForms::Form.find(params["bioguide_ids"].split)
+
+    if @message.submit
+      @name = user_params[:first_name] # for deliver_thanks_message
+      track_action
+      deliver_thanks_message
+      render partial: "tools/share"
+    else
+      render plain: I18n.t(:submission_failed, scope: :congress_forms), status: :bad_request
+    end
+  end
+
+  private
+
+  def set_congress_message_campaign
+    @campaign = CongressMessageCampaign.find(params["congress_message_campaign_id"])
+    @actionPage = @campaign.action_page
+    @action_page = @actionPage # Account for inconsistent naming in views.
+  end
+
+  def congress_message_params
+    # In Rails 5.1 we can do params.permit(common_attributes: {}, member_attributes: {})
+    params.permit.tap do |p|
+      p[:common_attributes] = params[:common_attributes] && params[:common_attributes].permit!
+      p[:member_attributes] = params[:member_attributes] && params[:member_attributes].permit!
+    end
+  end
+
+  def user_params
+    params.require(:common_attributes).permit.tap do |p|
+      common = params[:common_attributes]
```

Comment:
This mapping to and from Congress Forms-style field name formatting also happens in `CongressMessage::new_from_lookup`. It might be nice to do it all internally in that class and use friendlier field/param names out here. 

In general, I wasn't totally sure about the best way to represent filled in values there. I went with `common_attributes` and `member_attributes` to mirror the way `attributes` returns a hash, but I'm open to other ideas.